amd_tressfx_vulkan/src/TressFXRendererVulkan.cpp

```diff
@@ -0,0 +1,1569 @@
+        dsvDesc.viewType = VK_IMAGE_VIEW_TYPE_2D;
+        dsvDesc.subresourceRange.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+        dsvDesc.subresourceRange.layerCount = 1;
+        dsvDesc.subresourceRange.levelCount = 1;
+        dsvDesc.image = m_pSMHairTexture;
+        AMD_V_RETURN(vkCreateImageView(pvkDevice, &dsvDesc, nullptr, &m_pSMHairView));
+
+        VkImageMemoryBarrier defineHairSMLayoutBarrier[] = {getImageMemoryBarrier(
+            m_pSMHairTexture, 0, VK_ACCESS_SHADER_READ_BIT, VK_IMAGE_LAYOUT_UNDEFINED,
+            VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL, VK_IMAGE_ASPECT_DEPTH_BIT)};
+
+        vkCmdPipelineBarrier(commandBuffer, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT,
+                             VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT, 0, 0, nullptr, 0,
+                             nullptr, AMD_ARRAY_SIZE(defineHairSMLayoutBarrier),
+                             defineHairSMLayoutBarrier);
+    }
+
+    // Noise texture
+    {
+        VkImageCreateInfo noiseTextureInfo = getImageCreateInfo(
+            VK_FORMAT_R32G32B32A32_SFLOAT, 512, 512,
+            VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT);
+        AMD_V_RETURN(
+            vkCreateImage(pvkDevice, &noiseTextureInfo, nullptr, &m_pNoiseTexture));
+
+        VkMemoryRequirements mem_requirement;
+        vkGetImageMemoryRequirements(pvkDevice, m_pNoiseTexture, &mem_requirement);
+        VkMemoryAllocateInfo memoryAllocateInfo = {VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO,
+                                                   nullptr, mem_requirement.size,
+                                                   texture_memory_index};
+        AMD_V_RETURN(
+            vkAllocateMemory(pvkDevice, &memoryAllocateInfo, nullptr, &m_pNoiseMemory));
+        AMD_V_RETURN(vkBindImageMemory(pvkDevice, m_pNoiseTexture, m_pNoiseMemory, 0));
+
+        XMFLOAT4 *noiseArray = new XMFLOAT4[512 * 512];
+        for (UINT i = 0; i < 512 * 512; i++)
+        {
+            noiseArray[i].x = rand() / (float)RAND_MAX;
+            noiseArray[i].y = rand() / (float)RAND_MAX;
+            noiseArray[i].z = rand() / (float)RAND_MAX;
+            noiseArray[i].w = rand() / (float)RAND_MAX;
+        }
+
+        void *memoryPointer;
+        vkMapMemory(pvkDevice, scratchMemory, offsetInScratchBuffer,
+                    512 * 512 * sizeof(XMFLOAT4), 0, &memoryPointer);
+        memcpy(memoryPointer, noiseArray, 512 * 512 * sizeof(XMFLOAT4));
+        vkUnmapMemory(pvkDevice, scratchMemory);
+
+        VkImageMemoryBarrier makeNoiseTextureTransferableBarrier[] = {
+            getImageMemoryBarrier(m_pNoiseTexture, 0, VK_ACCESS_TRANSFER_WRITE_BIT,
+                                  VK_IMAGE_LAYOUT_UNDEFINED,
+                                  VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL)};
+
+        vkCmdPipelineBarrier(commandBuffer, VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT,
+                             VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT, 0, 0, nullptr, 0,
```

Comment:
Not 100% sure on this one, but shouldn't it be `VK_PIPELINE_STAGE_ALL_COMMANDS_BIT`? Bottom of pipe for the destination looks fishy.